pkg/operator/ceph/cluster/nodedaemon/keyring.go

```diff
@@ -72,15 +75,66 @@ func cephCrashCollectorKeyringCaps() []string {
 	}
 }
 
-func createCrashCollectorKeyring(s *keyring.SecretStore) (string, error) {
+func createCrashCollectorKeyring(s *keyring.SecretStore, context *clusterd.Context, clusterInfo *client.ClusterInfo) (string, error) {
 	key, err := s.GenerateKey(crashCollectorKeyringUsername, cephCrashCollectorKeyringCaps())
 	if err != nil {
 		return "", err
 	}
 
+	clusterObj := &cephv1.CephCluster{}
+	if err := context.Client.Get(clusterInfo.Context, clusterInfo.NamespacedName(), clusterObj); err != nil {
+		return "", errors.Wrapf(err, "failed to get cluster %v", clusterInfo.NamespacedName())
+	}
```

Comment:
I failed to catch this here but noticed it in the exporter PR. Since this is the CephCuster reconcile, there isn't a need to get the CephCluster again here when it can be passed in. 

Depending on what you'd prefer, could you address this in a new PR or in the exporter PR? Either one is fine by me, but avoiding truly unnecessary API calls would be good since we're already adding so many